graal/backends/core/codep/codep.py

```diff
@@ -0,0 +1,211 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#     inishchith <inishchith@gmail.com>
+#
+
+import logging
+import os
+from graal.backends.core.analyzer_composition_factory import AnalyzerCompositionFactory
+from graal.backends.core.codep.compositions.composition_reverse import CATEGORY_CODEP_PYREVERSE
+
+from graal.graal import (Graal,
+                         GraalCommand,
+                         GraalError,
+                         GraalRepository,
+                         DEFAULT_WORKTREE_PATH)
+from graal.backends.core.analyzers.analyzer import Analyzer
+from graal.backends.core.analyzers.jadolint import Jadolint, DEPENDENCIES
+from graal.backends.core.analyzers.reverse import Reverse
+from perceval.utils import DEFAULT_DATETIME, DEFAULT_LAST_DATETIME
+
+CATEGORY_PACKAGE = "graal.backends.core.codep.compositions"
+DEFAULT_CATEGORY = CATEGORY_CODEP_PYREVERSE
+
+logger = logging.getLogger(__name__)
+
+
+class CoDep(Graal):
+    """CoDep backend.
+
+    This class extends the Graal backend. It extract package and class dependencies
+    of a Python module to understand its evolution.
+
+    :param uri: URI of the Git repository
+    :param gitpath: path to the repository or to the log file
+    :param worktreepath: the directory where to store the working tree
+    :param exec_path: path of the executable to perform the analysis
+    :param entrypoint: the entrypoint of the analysis
+    :param in_paths: the target paths of the analysis
+    :param out_paths: the paths to be excluded from the analysis
+    :param details: if enable, it returns fine-grained results
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+
+    :raises RepositoryError: raised when there was an error cloning or
+        updating the repository.
+    """
+    version = '0.4.0'
+
+    def __init__(self, uri, git_path, worktreepath=DEFAULT_WORKTREE_PATH, exec_path=None,
+                 entrypoint=None, in_paths=None, out_paths=None, details=False,
+                 tag=None, archive=None):
+        super().__init__(uri, git_path, worktreepath, exec_path=exec_path,
+                         entrypoint=entrypoint, in_paths=in_paths, out_paths=out_paths, details=details,
+                         tag=tag, archive=archive)
+
+        self.__factory = AnalyzerCompositionFactory(CATEGORY_PACKAGE, jadolint_exec_path=exec_path)
```

Comment:
het passen van jadolint_exec_path is nu inconsistent met wat leon gedaan heeft vgm. die passt hem in de analyze method namelijk. Als jij hem daar ook naartoe doet, kan de kwargs ook weg uit de factory.